src/devices/imuBosch_BNO055/imuBosch_BNO055.cpp

```diff
@@ -675,6 +677,196 @@ bool BoschIMU::calibrate(int ch, double v)
     return false;
 }
 
+
+yarp::dev::MAS_status BoschIMU::genericGetStatus(size_t sens_index) const
+{
+    if (sens_index != 0)
+    {
+        return yarp::dev::MAS_status::MAS_ERROR;
+    }
+    
+    return yarp::dev::MAS_status::MAS_OK;
+}
+
+bool BoschIMU::genericGetSensorName(size_t sens_index, string& name) const
+{
+    if (sens_index != 0)
+    {
+        return false;
+    }
+    
+    name = m_sensorName;
+    return true;
+}
+
+bool BoschIMU::genericGetFrameName(size_t sens_index, string& frameName) const
+{
+    if (sens_index != 0)
+    {
+        return false;
+    }
+    
+    frameName = m_frameName;
+    return true;
+
+}
+
+size_t BoschIMU::getNrOfThreeAxisLinearAccelerometers() const
+{
+    return 1;
+}
+
+
+yarp::dev::MAS_status BoschIMU::getThreeAxisLinearAccelerometerStatus(size_t sens_index) const
+{
+    return genericGetStatus(sens_index);
+}
+
+bool BoschIMU::getThreeAxisLinearAccelerometerName(size_t sens_index, string& name) const
+{
+    return genericGetSensorName(sens_index, name);
+}
+
+bool BoschIMU::getThreeAxisLinearAccelerometerFrameName(size_t sens_index, string& frameName) const
+{
+    return genericGetFrameName(sens_index, frameName);
+}
+
+bool BoschIMU::getThreeAxisLinearAccelerometerMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const
+{
+    if (sens_index != 0)
+    {
+        return false;
+    }
+    
+    out.resize(3);
+    out.clear();
```

Comment:
If I am not wrong, this is equivalent to a `resize(0)`, so it is not correct.

Reply:
oh yeah! it had to be the reverse way!

Reply:
Fixed it. Thanks.

Reply:
I miss the point of the clear in any case. If you are going to resize the vector to size 3, and you are going to overwrite all its elements, what's the point of  clearing the vector before?

Reply:
Ok it's just my extra cautious compulsion of avoiding any malloc problems. But your point makes sense. It's just not needed at all, just increases the overhead. I will "clear()" it up.

Reply:
I checked and apparently `clear()` is not equivalent to `resize(0)`, is actually much worse w.r.t. to memory management: [`VectorOf::clear()`](https://github.com/robotology/yarp/blob/46119b13997f346a2fe0700993196952beabb6f1/src/libYARP_sig/include/yarp/sig/Vector.h#L472) calls  [`ManagedBytes::clear()`](https://github.com/robotology/yarp/blob/46119b13997f346a2fe0700993196952beabb6f1/src/libYARP_OS/src/ManagedBytes.cpp#L129) that always calls `delete`, meaning that the subsequent `resize(3)` will always call a `new`. [`vector.resize(vector.size())`](https://github.com/robotology/yarp/blob/46119b13997f346a2fe0700993196952beabb6f1/src/libYARP_sig/include/yarp/sig/Vector.h#L246) is not almost a no-op as I expected, but at least is not allocating again memory if not necessary.

Reply:
Fixed it. Thanks.